diff --git a/setuptools/_distutils/sysconfig.py b/setuptools/_distutils/sysconfig.py
index aae9c1b..469ed85 100644
--- a/setuptools/_distutils/sysconfig.py
+++ b/setuptools/_distutils/sysconfig.py
@@ -339,6 +339,9 @@ def customize_compiler(compiler):  # noqa: C901
         if 'RANLIB' in os.environ and compiler.executables.get('ranlib', None):
             compiler.set_executables(ranlib=os.environ['RANLIB'])

+        if 'RANLIB' in os.environ and 'ranlib' in compiler.executables:
+            compiler.set_executables(ranlib=os.environ['RANLIB'])
+
         compiler.shared_lib_extension = shlib_suffix

